tests/fsharpqa/Source/Conformance/SpecialAttributesAndTypes/Imported/CallerInfo/CallerMemberName.fs

```diff
@@ -0,0 +1,150 @@
+namespace Test
+
+open System.Runtime.CompilerServices
+open System.Reflection
+open CSharpLib
+
+[<assembly:MyCallerMemberNameAttribute>]
+do
+    ()
+
+[<MyCallerMemberName()>]
+type MyTy() =
+    let functionVal = MyTy.GetCallerMemberName
+    let typeLetValue = MyTy.GetCallerMemberName()
+    let typeLetFunc (i:int) = i, MyTy.GetCallerMemberName()
+    let typeLetFuncNested () =
+        let nestedFunc () = MyTy.GetCallerMemberName()
+        nestedFunc ()
+    do
+        MyTy.Check(MyTy.GetCallerMemberName(), Some(".ctor"), "primary ctor")
+
+    new(i : int) =
+        MyTy.Check(MyTy.GetCallerMemberName(), Some(".ctor"), ".NET ctor")
+        MyTy()
+    
+    member __.Item
+        with get(i:int) = MyTy.GetCallerMemberName()
+        and set(i:int) (v:string option) =
+            MyTy.Check(MyTy.GetCallerMemberName(), Some("Item"), "index setter")
+
+    member __.CheckMembers() =
+        MyTy.Check(MyTy.GetCallerMemberName(), Some("CheckMembers"), ".NET method")
+        MyTy.Check(typeLetValue, Some("typeLetValue"), "type let value")
+        MyTy.Check(typeLetFunc 2 |> snd, Some("typeLetFunc"), "type let func")
+        MyTy.Check((typeLetFuncNested ()) , Some("typeLetFuncNested"), "type let func nested")
+        MyTy.Check(__.GetCallerMemberNameProperty1, Some("GetCallerMemberNameProperty1@"), "auto property getter")
+        MyTy.Check(MyTy.GetCallerMemberNameProperty, Some("GetCallerMemberNameProperty"), "property getter")
+        MyTy.GetCallerMemberNameProperty <- Some("test")
+        MyTy.Check(__.[10], Some("Item"), "indexer getter")
+        __.[10] <- Some("test")
+
+        let result =
+            [1..10]
+            |> List.map (fun i -> MyTy.GetCallerMemberName())
+            |> List.head
+        MyTy.Check(result, Some("CheckMembers"), "lambda")
+        MyTy.Check(functionVal (), Some("functionVal"), "functionVal")
+        ()
+
+    static member GetCallerMemberName([<CallerMemberName>] ?memberName : string) =
+        memberName
+
+    static member Check(actual : string option, expected : string option, message) =
+        printfn "%A" actual
+        if actual <> expected then
+            failwith message
+    
+    static member GetCallerMemberNameProperty
+        with get () = MyTy.GetCallerMemberName()
+        and set (v : string option) =
+            MyTy.Check(MyTy.GetCallerMemberName(), Some("GetCallerMemberNameProperty"), "property setter")
+        
+    member val GetCallerMemberNameProperty1 = MyTy.GetCallerMemberName() with get, set
+
+[<Struct>]
+type MyStruct =
+    val A : int
+    new(a : int) =
+        { A = a }
+        then
+            MyTy.Check(MyTy.GetCallerMemberName(), Some(".ctor"), "struct ctor")
```

Comment:
yes I checked it and also look at this test https://github.com/dotnet/roslyn/blob/56f605c41915317ccdb925f66974ee52282609e7/src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_CallerInfoAttributes.cs#L1274